internal/provider/resources/user.go

```diff
@@ -0,0 +1,243 @@
+				CustomType:  customtypes.UUIDType{},
+			},
+			"handle": schema.StringAttribute{
+				Description: "A unique handle for the user, containing only lowercase letters, numbers, and dashes.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"first_name": schema.StringAttribute{
+				Description: "First name of the user",
+				Computed:    true,
+				Optional:    true,
+			},
+			"last_name": schema.StringAttribute{
+				Description: "Last name of the user",
+				Computed:    true,
+				Optional:    true,
+			},
+			"email": schema.StringAttribute{
+				Description: "Email of the user",
+				Computed:    true,
+				Optional:    true,
+			},
+		},
+	}
+}
+
+// Create creates a new user.
+func (r *UserResource) Create(_ context.Context, _ resource.CreateRequest, resp *resource.CreateResponse) {
+	resp.Diagnostics.AddError("Not implemented", "Creating a user is not yet implemented")
+}
+
+// Read reads a user.
+func (r *UserResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state UserResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if state.ID.IsNull() {
+		resp.Diagnostics.AddError(
+			"ID is unset",
+			"Ensure the user has been imported into the state before attempting to read.",
+		)
+
+		return
+	}
+
+	client, err := r.client.Users()
+	if err != nil {
+		resp.Diagnostics.Append(helpers.CreateClientErrorDiagnostic("User", err))
+
+		return
+	}
+
+	user, err := client.Read(ctx, state.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("User", "read", err))
+
+		return
+	}
+
+	copyUserToModel(user, &state)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
+}
+
+// Update updates a user.
+func (r *UserResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan UserResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := r.client.Users()
+	if err != nil {
+		resp.Diagnostics.Append(helpers.CreateClientErrorDiagnostic("User", err))
+
+		return
+	}
+
+	err = client.Update(ctx, plan.ID.ValueString(), api.UserUpdate{
+		Handle:    plan.Handle.ValueString(),
+		FirstName: plan.FirstName.ValueString(),
+		LastName:  plan.LastName.ValueString(),
+		Email:     plan.Email.ValueString(),
+	})
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("User", "update", err))
+
+		return
+	}
+
+	user, err := client.Read(ctx, plan.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("User", "read", err))
+
+		return
+	}
+
+	copyUserToModel(user, &plan)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+// Delete deletes a user.
+func (r *UserResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
```

Comment:
Thinking out loud: should we instead _not_ implement Delete? I don't think an end user would ever truly want to delete a User if they remove the configuration from Terraform. Deleting users is, as far as I know, a more involved process that also includes removing API keys, memberships, workspaces, personal accounts, etc.

If we did go that way, we could also remove the lifecycle deletion protection on the acceptance tests.

Reply:
im actually not sure of the answer - im assuming that a valid DELETE endpoint exists

however, i'd advocate for omitting DELETE in tandem with CREATE - they should mirror each other in terms of what we allow the provider to manage.  in our case, we omit CREATE, so I would say we may want to omit DELETE to avoid the issue you raised

Reply:
That sounds good, updated in 8bae2dd.